dask/highlevelgraph.py

```diff
@@ -160,6 +160,22 @@ def map_tasks(self, func: Callable[[Iterable], Iterable]) -> "Layer":
 
         return BasicLayer({k: func(v) for k, v in self.items()})
 
+    def clone(self, all_external_keys):
+        from distributed.utils_comm import subs_multiple
+
+        dsk_new = {}
+        old_to_new_keys = {}
+        for key, value in self.items():
+            new_key = clone_key(key)
+            old_to_new_keys[key] = new_key
+            dsk_new[new_key] = value
+        # Include external keys in task substitutions
+        for key in all_external_keys:
+            old_to_new_keys[key] = clone_key(key)
+
+        dsk_new = subs_multiple(dsk_new, old_to_new_keys)
+        return BasicLayer(dsk_new)
```

Comment:
We will eventually want to handle different layers differently, but I agree that it makes sense to start with something simple like this to make sure that we get the feel right.

Reply:
Yeah, I wanted to start simple and then overload `.clone()` on `Layer` subclasses as needed